src/coreclr/crossgen-corelib.proj

```diff
@@ -3,21 +3,107 @@
   <Import Project="Directory.Build.targets" />
 
   <Target Name="Build">
-    <ItemGroup>
-      <_CoreClrBuildArg Condition="'$(TargetArchitecture)' != ''" Include="-$(TargetArchitecture)" />
-      <_CoreClrBuildArg Include="-$(Configuration.ToLower())" />
-      <_CoreClrBuildArg Condition="!$([MSBuild]::IsOsPlatform(Windows))" Include="-os $(TargetOS)" />
-      <_CoreClrBuildArg Condition="'$(CrossBuild)' == 'true'" Include="-cross" />
-      <_CoreClrBuildArg Condition="'$(PortableBuild)' != 'true'" Include="-portablebuild=false" />
-    </ItemGroup>
+    <PropertyGroup>
+      <!-- When the 'Crossgen1' or 'Crossgen2' environment variable it set to 'true', -->
+      <!-- it will force the script to use the specified compiler to build System.Private.CoreLib. -->
+      <!-- In absence of these variables, the script uses its own default. -->
+      <UseCrossgen2ByDefault>true</UseCrossgen2ByDefault>
+
+      <UseCrossgen2 Condition="'$(Crossgen1)' == 'true'">false</UseCrossgen2>
+      <UseCrossgen2 Condition="'$(Crossgen2)' == 'true'">true</UseCrossgen2>
+      <UseCrossgen2 Condition="'$(UseCrossgen2)' == ''">$(UseCrossgen2ByDefault)</UseCrossgen2>
```

Comment:
Why do we need two properties, `Crossgen1` and `Crossgen2`, instead of a single one (e.g., `UseCrossgen2` with values `true`, `false`, and undefined)?

Reply:
This is just an initial attempt at build configurability. I'm open to any suggestions, my initial expectation was that during the gradual switch-over each script will have its own status and timeline for moving to Crossgen2 so that in general every script should have its own default plus a way to externally override its behavior by forcing the use of CG1 / CG2. Using "Crossgen1" and "Crossgen2" as the outside-facing environment variables seemed like an easy option to me but I'll welcome alternate suggestions.

Reply:
Could you help me understand how having two separate properties would be better than the single `UseCrossgen2` property, which you can default to either `true` or `false` (depending on the script) and still can override from the command line with `/p:UseCrossgen2=...`?
```xml
<!-- Set the default value for the given script, which may be true of false -->
<UseCrossgen2 Condition="'$(UseCrossgen2)' == ''">true</UseCrossgen2>
```

Reply:
Thank you for the changes!